test/test.js

```diff
@@ -46,6 +46,44 @@ it('thenifyAll(fs, destination)', function () {
   })
 })
 
+it('thenifyAll(user, ["show"]) should maintain the receiver', function () {
+  var user = Object.create({
+    show: function (cb) {
+      cb(null, this.name);
+    }
+  }, {
+    name: {
+      value: 'test'
+    }
+  });
+
+  return thenify(user, ['show']).show().then(function (name) {
+    assert.equal(name, 'test')
+  })
+})
+
+it('thenifyAll(user) should maintain the receiver', function () {
+  var user = {
+    name: 'test',
+    load: function () {
+      var cb = arguments[arguments.length - 1]
+      cb(null, this.name)
+    },
+    show: function (cb) {
+      this.load(cb)
```

Comment:
For this case, if not bind to `source`, show will not resolved or rejected. Because of `load` get arguments with two `cb`, first is passed by `show`, second is by `load`'s promisify. So, bind to `source`       
to use origin `load`.

Anyway, it's weird case. haha!